tests/kernel/common/src/errno.c

```diff
@@ -98,4 +98,61 @@ void test_thread_context(void)
 	if (errno != errno_values[N_THREADS]) {
 		rv = TC_FAIL;
 	}
+
+	/* Make sure all the test thread end. */
+	for (int ii = 0; ii < N_THREADS; ii++) {
+		k_thread_join(&threads[ii], K_FOREVER);
+	}
+}
+
+
+#define ERROR_ANY 0xfc
+
+void thread_entry_user(void *p1, void *p2, void *p3)
+{
+#ifdef CONFIG_ARCH_POSIX
+	/* The errno in native posix will be handled by native
+	 * operation system, so we skip it.
+	 */
+	ztest_test_skip();
+#else
+	int got_errno;
+
+	/* assign the error number to C standard errno */
+	errno = ERROR_ANY;
+
+	/* got errno zephyr stored */
+	got_errno = *(z_errno());
+
+	zassert_equal(errno, got_errno, "errno is not corresponding.");
+#endif
+}
+
+/**
+ * @brief Verify errno works well
+ *
+ * @details Check whether assign a C standard errno is successfully
+ * stored by zephyr, no matter it is using tls or not.
+ *
+ * @ingroup kernel_threadcontext_tests
+ */
+void test_errno(void)
+{
+	k_tid_t tid;
+
+	tid = k_thread_create(&threads[0], stacks[0], STACK_SIZE,
+				thread_entry_user, NULL, NULL, NULL,
+				K_PRIO_PREEMPT(1), K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+
+#ifdef CONFIG_USERSPACE
```

Comment:
why are you creating the thread twice? just create one time with K_USER and grant access in the main thread, also to test userspace you should use ztest_user_unit_test(..)

Reply:
Got it, I am working on it now, thanks for your review!

Reply:
Got it fixed, could you please help to review again, thanks!